packages/superset-ui-legacy-plugin-chart-table/src/ReactDataTable.tsx

```diff
@@ -0,0 +1,252 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { t } from '@superset-ui/translation';
+import React, { useEffect, createRef } from 'react';
+import { formatNumber, NumberFormats } from '@superset-ui/number-format';
+import { getTimeFormatter } from '@superset-ui/time-format';
+import { filterXSS } from 'xss';
+
+// initialize datatables.net
+import $ from 'jquery';
+import dt from 'datatables.net-bs/js/dataTables.bootstrap';
+import 'datatables.net-bs/css/dataTables.bootstrap.css';
+import './Table.css';
+
+import { DataTableProps } from './transformProps';
+
+// Depending on how the modules are imported, `dt` may be a CommonJS init function,
+// or the DataTable class itself. In case it is the former, we'd need to tell it
+// where is jQuery.
+if (!dt.$) {
+  dt(window, $);
+}
+
+const { PERCENT_3_POINT } = NumberFormats;
+const isPontiallyHTML = (text: string) => /<[^>]+>/.test(text);
```

Comment:
* typo `isPontiallyHTML` => `isPartiallyHTML`. 
* The regex can be defined outside of the function to avoid parsing regex every function call.

Reply:
- Thanks for noticing the typo! I actually meant `isPotentiallyHTML`, which probably makes more sense than "partially"?
- I was hoping modern JS engines can already optimize for RegExp literals.

Perf test seems to suggest the uncached approach is actually faster:
https://www.measurethat.net/Benchmarks/Show/7486/1/regexp-literal-cache-or-not

I bet it has something to do with variable scope resolution.